Code.gs

```diff
@@ -178,46 +171,82 @@ function main(){
     }
   }
   //----------------------------------------------------------------
-
-
-
-  //-------------- Remove Or modify events from calendar -----------  
+  
+  
+  //-------------- Remove Or modify events from calendar -----------
+  var alreadyProcessedFids = [];
   Logger.log("Checking " + calendarEvents.length + " events for removal or modification");
   for (var i = 0; i < calendarEvents.length; i++){
     var tagValue = calendarEvents[i].getTag("FID");
     var feedIndex = feedEventIds.indexOf(tagValue);
     
+    if (alreadyProcessedFids.indexOf(tagValue) > -1)
+      continue;
+    
     if(removeEventsFromCalendar){
       if(feedIndex  == -1 && tagValue != null){
-        Logger.log("    Deleting " + calendarEvents[i].getTitle());
-        calendarEvents[i].deleteEvent();
+        Logger.log("    Deleting " + calendarEvents[i].getTitle() + " (id: " + tagValue + ")");
+        calendarEvents[i].getEventSeries().deleteEventSeries(); //Delete the event by series. This works even if it is not a recurring event 
       }
     }
 
     if(modifyExistingEvents){
       if(feedIndex != -1){
-        var e = calendarEvents[i];
-        var fes = events.filter(sameEvent, calendarEvents[i].getTag("FID"));
+        Logger.log("    Checking for modification " + calendarEvents[i].getTitle() + " (id: " + tagValue + ")");
         
-        if(fes.length > 0){
-          var fe = fes[0];
-
+        var fes = icsEvents.filter(sameEvent, calendarEvents[i].getTag("FID"));
+        
+        if (fes.length <= 0)
+          continue;
+          
+        var fe = fes[0];
+        
+        if (fe.recurrence != null){
+          var eSeries = calendarEvents[i].getEventSeries();
+          
+          //Since there is no CalendarEventSeries.getRecurrence() method we can use to check then we will always set the recurrence
+          //(which also means setting the time). This also solves the situation of a regular event becoming a recurring event.
+          eSeries.setRecurrence(fe.recurrence, fe.startTime, fe.endTime);
```

Comment:
setRecurrence does not delete the exisiting recurring events.
So if you change an event from Interval 2 days to interval 3 days you will end up with it repeating every 2 and every 3 days..Might have to delete the old Eventseries and freshly create the new one.